contracts/staking/ERC20TokenStakingManager.sol

```diff
@@ -107,7 +107,7 @@ contract ERC20TokenStakingManager is
     function initializeDelegatorRegistration(
         bytes32 validationID,
         uint256 delegationAmount
-    ) external returns (bytes32) {
+    ) external nonReentrant returns (bytes32) {
```

Comment:
For my own understanding, is there an advantage to moving `nonReentrant` here rather than in `_initializeDelegatorRegistration`?

Reply:
The idiomatic pattern is to place `nonReentrant` on all external functions. That way a contract can never be called back into by unknown code in an external contract, but can still call its own internal functions without worry.

Reply:
Makes sense, thanks for the explanation.